modules/core/src/encoder/field/PurposeRestrictionVectorEncoder.ts

```diff
@@ -14,19 +14,28 @@ export class PurposeRestrictionVectorEncoder {
     // if the vector is empty we'll just return a string with just the numRestricitons being 0
     if (!prVector.isEmpty()) {
 
-      const nextGvlVendor = (vendorId, lastVendorId): number => {
+      const gvlVendorIds = Array.from(prVector.gvl.vendorIds);
 
-        for (let nextId = vendorId + 1; nextId <= lastVendorId; nextId++) {
+      const nextGvlVendor = (vendorId, lastVendorId) => {
 
-          if (prVector.gvl.vendorIds.has(nextId)) {
+        const firstIndex = gvlVendorIds.indexOf(vendorId);
+        const lastIndex = gvlVendorIds.indexOf(lastVendorId);
 
-            return nextId;
+        if (lastIndex - firstIndex > 0) {
```

Comment:
Could this one be true if first index is equal -1 -> not found?

Reply:
Yes that condition is required only to check if we are at the "end" of the used GVL vendor list in fact in line 26 you see we are taking the nextIndex the one that is specified in the GVL vendor list. 
This makes possible to skip every extra loop in the purpose restriction encode process.